docs_src/getting-started/native/Ch-BuildRunOnLinuxDistro.md

```diff
@@ -0,0 +1,376 @@
+    - See [How to install and configure Redis on Ubuntu 20.04](https://linuxize.com/post/how-to-install-and-configure-redis-on-ubuntu-20-04/)
+    - How to check for existence and version on your machine
+
+        ![image](RedisCheck.png)
+
+    - Your installation process may vary based on Linux version/distribution
+
+- ZeroMQ
+    - See [this guide for a script to install ZMQ](https://gist.github.com/katopz/8b766a5cb0ca96c816658e9407e83d00)
+    - How to check for existence and version on your machine
+
+        ![image](ZeroMQCheck.png)
+
+- Git
+    - Git is already installed with Ubuntu 20.04
+    - If not already provided with your OS, see [Install Git on Linux](https://www.atlassian.com/git/tutorials/install-git#linux)
+    - How to check for existence and version on your machine
+
+        ![image](GitCheck.png)
+
+## Prepare your environment
+
+In this guide, you will be building and running EdgeX in "non-secure" mode.  That is, you will be building and running the EdgeX platform without the security services and security configuration.  An environmental variable, `EDGEX_SECURITY_SECRET_STORE`,  is set to indicate whether the EdgeX services are expected to initialize and use the secure secret store.  By default, this variable is set to `true`.  Prior to building and running EdgeX, set this environment variable to false.
+
+``` Shell
+export EDGEX_SECURITY_SECRET_STORE=false 
+```
+
+This can be done in the terminal from which you build and run EdgeX or you can set it in your user's profile to make an environment persist across terminal sessions.  See [How to Set Environment Variables in Linux](https://www.serverlab.ca/tutorials/linux/administration-linux/how-to-set-environment-variables-in-linux/) for assistance.
+
+## Download EdgeX Source
+
+In order to build and run EdgeX micro services, you will first need to get the source code for the platform.  Using git, clone the EdgeX repositories with the following commands:
+
+!!! Tip
+    You may wish to create a new folder and then issue these git commands from that folder so that all EdgeX code is neatly stored in a single folder.
+
+``` Shell
+git clone https://github.com/edgexfoundry/edgex-go.git
+git clone https://github.com/edgexfoundry/device-virtual-go.git
+git clone https://github.com/edgexfoundry/app-service-configurable.git
+git clone https://github.com/lf-edge/ekuiper.git
+git clone https://github.com/edgexfoundry/edgex-ui-go.git
+```
+
+Note that a new folder, named for the repository, gets created containing source code with each of the git clones above.
+
+!!! Warning
+    These git clone operations pull from the main branch of the EdgeX repositories.  This is the current working branch in EdgeX development.  See the [git clone documentation](https://git-scm.com/docs/git-clone) for how to clone a specific named release or branch.
+
+## Build EdgeX Services
+
+With the source code, you can now build the EdgeX services, GUI, as well as eKuiper rules engine.  
```

Comment:
I'll mention the availability of binaries.  The fact that we provide the build for eKuiper too is good, I think, since people can do everything from scratch.  I also found that their binaries don't come with the edgex elements (the `build_with_edgex`).